conda/libmesh/build.sh

```diff
@@ -66,12 +66,14 @@ make -j $CORES
 make install
 sed_replace
 
-# Set LIBMESH_DIR environment variable for those that need it
+# Set LIBMESH_DIR, Eigen3_DIR
 mkdir -p "${PREFIX}/etc/conda/activate.d" "${PREFIX}/etc/conda/deactivate.d"
 cat <<EOF > "${PREFIX}/etc/conda/activate.d/activate_${PKG_NAME}.sh"
 export LIBMESH_DIR=${PREFIX}/libmesh
+export Eigen3_DIR=${PREFIX}/libmesh/include/Eigen
```

Comment:
Does there also need to be a copy of the `$CONDA_PREFIX_1/pkgs/eigen-3.4.0-h940c156_0/share/eigen3/cmake/*` files into this `Eigen3_DIR`?

Reply:
CONDA_PREFIX_1 does not always exist. But more importantly what you see here are internal `conda build` environment variables needed during package creation only. `${PREFIX}` gets replaced with a special `placeholder` tag, that Conda understands to replace _that_ with the installation location of your Conda implementation environments:

```bash
bash Mambaforge-Linux-x86_64.sh -b -p ~/mambaforge3
                                      ^^^^^^^^^^^^ this location can be anything and ends up being
                                                   the absolute path replacing 'placeholder'. e.g:

mamba create -n moose moose-libmesh

placeholder = /Users/you/mambaforge3/envs/moose/
```
It is this neat feature, that makes Conda relocatable and installable to any location you have write access (we love it, because we are able to tell folks they don't need sudo to install a MOOSE developers environment).

All that aside, are there additional environment variables I am forgetting about?

Reply:
Perhaps I'm still misunderstanding something. I'm trying to express something @GiudGiud mentioned on Slack.

DAGMC will error unless it finds the following CMake files inside of `$Eigen3_DIR`:

```
(moose) {12:01}~/projects/cardinal:  ➭ ls $Eigen3_DIR
Eigen3Config.cmake        Eigen3ConfigVersion.cmake Eigen3Targets.cmake       UseEigen3.cmake
```

Will those files be found inside of `$Eigen3_DIR` (which conda will now define as you've described above)?

Reply:
These concerns were discussed in Slack, and resolved. The above changes are correct.